source/agora/consensus/EnrollmentManager.d

```diff
@@ -711,6 +718,24 @@ public class EnrollmentManager
         }
     }
 
+    ///
+    private void collectValidatorStats (Collector collector)
+    {
+        auto validator_count = validator_set.count();
+        if(!validator_count)
+            validator_stats.setAgoraValidatorsGaugeTo(validator_count);
+        ulong preimage_count;
+        Hash[] keys;
+        if (getEnrolledUTXOs(keys))
+        {
+            foreach(const ref key; keys)
+                preimage_count += validator_set.getPreimage(key).distance;
```

Comment:
Hmm.. This is a bit of a strange metric. If the validator re-enrolls then the distance becomes zero again, so the stats will end up varying a lot based on when they are emitted, and also based on when each validator has enrolled.

We should probably think about what would be a more useful statistic w.r.t. preimages. @Geod24 do you have any ideas for a good metric?

Reply:
Umm ok. I think we'll have to wait and see. My prediction is this output won't be very useful. :>

For me it would be more useful to have this stat per-node. But let's go ahead with this for now.

Reply:
Right, I missed this. @ferencdg : The `key` should be a dimension.